metric/instrument/instrument.go

```diff
@@ -64,27 +62,27 @@ func (o descOpt) applyInt64Observer(c Int64ObserverConfig) Int64ObserverConfig {
 // WithDescription sets the instrument description.
 func WithDescription(desc string) Option { return descOpt(desc) }
 
-type unitOpt unit.Unit
+type unitOpt string
 
 func (o unitOpt) applyFloat64(c Float64Config) Float64Config {
-	c.unit = unit.Unit(o)
+	c.unit = string(o)
 	return c
 }
 
 func (o unitOpt) applyInt64(c Int64Config) Int64Config {
-	c.unit = unit.Unit(o)
+	c.unit = string(o)
 	return c
 }
 
 func (o unitOpt) applyFloat64Observer(c Float64ObserverConfig) Float64ObserverConfig {
-	c.unit = unit.Unit(o)
+	c.unit = string(o)
 	return c
 }
 
 func (o unitOpt) applyInt64Observer(c Int64ObserverConfig) Int64ObserverConfig {
-	c.unit = unit.Unit(o)
+	c.unit = string(o)
 	return c
 }
 
 // WithUnit sets the instrument unit.
-func WithUnit(u unit.Unit) Option { return unitOpt(u) }
+func WithUnit(u string) Option { return unitOpt(u) }
```

Comment:
According to the [specs](https://opentelemetry.io/docs/reference/specification/metrics/api/#instrument-unit), the unit field can have a maximum length of 63 characters. 

> It can have a maximum length of 63 characters. The number 63 is chosen to allow the unit strings (including the \0 terminator on certain language runtimes) to be stored and compared as fixed size array/struct when performance is critical.

I've checked [Python](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/src/opentelemetry/metrics/_internal/instrument.py#L42) and [Java](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/internal/ValidationUtil.java#L90) implementations, it seems they have this validation on their end, is that something intentionally ignored in this implementation?

Reply:
We do not validate the unit, in compliance with [the metric API specification](https://github.com/open-telemetry/opentelemetry-specification/blob/361c9491d1a8c66757739d533162aee68272f629/specification/metrics/api.md#synchronous-instrument-api):

> [...] the API MUST accept a case-sensitive string that supports ASCII character encoding and can hold at least 63 characters. The API SHOULD NOT validate the `unit`.